### ISSUE TITLE
508-defect-2: Undescriptive headings, focus management, and more 

### ISSUE THREAD

**Issue by joshkimux** [department-of-veterans-affairs/va.gov-team#53471]
### Point of contact

@joshkimux

### Severity level

2, Serious. Should be fixed in 1-2 sprints post-launch.

### Details

[As documented in 10-10ez research](https://github.com/department-of-veterans-affairs/va.gov-research-repository/issues/154#issuecomment-1225813166), focus management in longer forms where a user may be stuck on the same step for _multiple_ pages is a confusing experience for screen reader users.

> P: See I don't understand 1 of 6. I already entered my name. Unless it's back up there. M: So you're saying, you already entered your name, why is on step 1? Did I capture that right? P: No, I entered my name, then it dropped me to ssn, but it's still saying step 1 of 6, so I don't know if it's actually changed the page or if my continue button just didn't work.

Several key issues exist here that wrap around one another:
- The language "step" implies one _unique_ page, which breaks down when multiple pages can exist within the same "step"
- Since we have a generic `h1` (which in and of itself is an a11y issue), it was decided to send the focus to the `h2` "step"; this was the best choice made at the time originally with the Section 508 office as
  - we didn't have access yet to user research with disabled assistive technology users
  - this was done before we gradually started breaking down longer steps into sub-steps (a la one thing per page), meaning there was less of a chance that there could be 3-5+ pages per step (at the cost of all the consequences that come with longer, more complex pages).
- There's often no other "unique" headings we can send focus to on a page as they're usually the legend/labels below the "step" `h2`

<img width="552" alt="Screenshot 2023-02-10 at 8 58 33 AM" src="https://user-images.githubusercontent.com/14154792/218109869-b69a3ade-275c-4f6a-ab7e-8377ee300951.png">

[More on this can be found in this design system issue ticket.](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/1538)

### Reproduction steps

1. Turn on your screen reader
2. Navigate using continue and back through the pages, confirm that the same h2 repeats every time and fails to be unique and descriptive after the first time it is announced."

### Proposed solution or next steps

[As documented in the design system issue ticket](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/1538), I recommend that our team work towards the "next" solution to serve as an interim until a simplified form header pattern can be created.

#### Add unique `h3`s to receive focus for forms in progress that will be published _before_ a simplified header is introduced
The ideal screen reader experience would entail hearing a unique topic/title/heading each time one navigates to a new page describing what that new page will cover. Although introducing the language change will reduce confusion to a significant degree, it still doesn't address the key a11y issue at the heart of it: generic headings/page titles that don't accomplish the ideal experience described.

To set up forms in progress to transition to a simplified header where the `h1` is a unique title to the page that takes focus (instead of a generic description of where you generally are in the form), we can explore temporarily adding `h3`s that uniquely describe a page in a form.

#### Acceptance criteria:
- [ ] A unique `h3` exists for every page within the form flow
  - [ ] [The `h3`s follow the language written on the testing plan page of this spreadsheet ](https://docs.google.com/spreadsheets/d/157gfgClyIoIMGrQky-KKvtaoug-Z5TSbBIguWAyr8zg/edit?usp=sharing)
  - [ ] Upon changing, mark the "no" to "yes" within the spreadsheet
- [ ] Focus goes to the unique `h3` when navigating between all pages within the form
- [ ] Focus does _not_ go to any foreign `h3`s that don't describe the purpose of the page (e.g. alerts)

### Type of issue

- [X] Screenreader
- [ ] Keyboard
- [X] Focus
- [ ] Headings
- [ ] Color
- [ ] Zoom
- [ ] Semantics
- [ ] Axe-core
- [ ] Something else

**Comment by saderagsdale** [comment]
@Mottie will ping platform support

**Comment by Mottie** [comment]
Work merged in and is now available in staging for testing.

**Comment by joshkimux** [comment]
**Update on this:**
- h3s are taking focus 💯 
- I need to do deeper screen reader testing on NVDA/Chrome and iOS VoiceOver to look into verbosity
- There are some bugs here and there with react fumbling the focus (I think)
 
Will update this ticket once I finish screen reader testing (which is the latter half of my review)

**Comment by joshkimux** [comment]
Just completed testing on iOS VoiceOver and NVDA and it's performing extraordinarily well 🥳 

Closing this out as it solves the root issue of
- undescriptive headings
- focus management

There may be new tickets opened up in case there's some unpredictable verbosity/bugs, but they wouldn't fall within the scope of this issue.

Huge, huge, huge shoutout to @Mottie for this effort; literally paving the way to fix the most wicked of accessibility problems in our form system, truly the 🐐
